File: app/src/main/java/org/schabi/newpipe/fragments/detail/StackItem.java
package org.schabi.newpipe.fragments.detail;

import org.schabi.newpipe.player.playqueue.PlayQueue;

import java.io.Serializable;

class StackItem implements Serializable {
    private final int serviceId;
    private String url;
    private String title;
    private PlayQueue playQueue;

    StackItem(final int serviceId, final String url,
              final String title, final PlayQueue playQueue) {
        this.serviceId = serviceId;
        this.url = url;
        this.title = title;
        this.playQueue = playQueue;
    }

    public void setUrl(final String url) {
        this.url = url;
    }

    public void setPlayQueue(final PlayQueue queue) {
        this.playQueue = queue;
    }

    public int getServiceId() {
        return serviceId;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(final String title) {
        this.title = title;
    }

    public String getUrl() {
        return url;
    }

    public PlayQueue getPlayQueue() {
        return playQueue;
    }

    @Override
    public String toString() {
        return getServiceId() + ":" + getUrl() + " > " + getTitle();
    }
}

File: app/src/main/java/org/schabi/newpipe/fragments/detail/StackItem.kt
package org.schabi.newpipe.fragments.detail

import org.schabi.newpipe.player.playqueue.PlayQueue
import java.io.Serializable

internal class StackItem(private val serviceId: Int, private var url: String?,
                         private var title: String, private var playQueue: PlayQueue?) : Serializable {
    fun setUrl(url: String?) {
        this.url = url
    }

    fun setPlayQueue(queue: PlayQueue?) {
        playQueue = queue
    }

    fun getServiceId(): Int {
        return serviceId
    }

    fun getTitle(): String? {
        return title
    }

    fun setTitle(title: String) {
        this.title = title
    }

    fun getUrl(): String? {
        return url
    }

    fun getPlayQueue(): PlayQueue? {
        return playQueue
    }

    public override fun toString(): String {
        return getServiceId().toString() + ":" + getUrl() + " > " + getTitle()
    }
}
